### ISSUE TITLE
Cacti 1.2.2: Multi Poller: Cannot create RRA / graphs

### ISSUE THREAD

**Issue by stefanbrudny** [Cacti/cacti#2556]
**Describe the bug**

No matter what combination I use nginx / apache / Gentoo / Ubuntu I can't have the graphs working on any of Remote Pollers. I keep getting:

On Ubuntu, Apache:

2019/03/22  00:46:19 - CMDPHP PHP ERROR NOTICE Backtrace:   (/graphs.php[117]:graph_edit(),  /graphs.php[1516]:draw_nontemplated_fields_graph_item(),  /lib/html_form_template.php[250]:draw_edit_form(),  /lib/html_form.php[112]:draw_edit_control(),  /lib/html_form.php[445]:CactiErrorHandler())
--
2019/03/22 00:46:19 - ERROR PHP NOTICE: Undefined index: value in file: /var/www/cacti/lib/html_form.php  on line: 445

On Central Cacti no real errors. 

It should not be relevant, but still: two ip namespace dmains are used, I substituted poller.php with a version which is not changed by cacti, to accommodate different spine configs between remote pollers.

That said, spine seems to work perfectly on all hosts: it collects results on all pollers and main site but rra won't get create.

Meanwhile Central Cacti works well and graphs are presented just fine.

**To Reproduce**
I wouldn't say its 100% bug, maybe its misconfiguration. However, considering amount of time spent in this is 99% a bug :-/

![image](https://user-images.githubusercontent.com/6334806/54792118-11fcb180-4c3d-11e9-86b8-b7586e252802.png)

**Comment by netniV** [comment]
This seems to be more a graph issue since it is attempting to draw items relating to a graph.  If this is only affecting certain graphs, make sure that they are in sync with the template they are based on.

**Comment by stefanbrudny** [comment]
I wouldn't say so. Installations are very similar. All graphs are working just fine on Central Cacti, whereas same templates and even simplest graphs do not on remote sites. I used one central, 2 in namespaces, and one another standalone just to make sure I am doing all according to documentation... no, wait... ;-)

**Comment by netniV** [comment]
If you have an error at line 445, you should have also had one for line 443, yet you haven't reported that.  Add some logging

```php
if (!isset($field_array['value'])) {
	cacti_log('bad field: ' . var_export($field_array, true));
}
```

**Comment by netniV** [comment]
This may be a plugin related issue, not a core issue since it relates to a field definition and we should have heard about a bad field definition from lots of people by now.

If you do manage to add the above lines, the entry in the log file will give us a breakdown on the contents of the field that should point us in the right direction of what is causing it.  

Additionally, If it seems useful enough, I may actually add it with some proper logging for the next release if so.

**Comment by stefanbrudny** [comment]
Gimme some time to nail it, I do not install these systems to monitor any of my installs, as it does fit for duty yet, and I do it just for fun (though I am now REALLY fed up with blocking).

I suspect no one have really gone that deep with assumptions taken from real environments, most of users tolerate deficencies. I never seen on cacti forum large installations thought, there might be some really well hidden.

**Comment by netniV** [comment]
The issue is that plugins can manipulate those field settings, plus we sometimes get things wrong.  That's why I do think if this debug code helps, i'll flesh it out more (plus it should not try and manipulate a value that doesn't even exist).

**Comment by stefanbrudny** [comment]
I thing I am closer. Indeed there are no graph template data items for the graph templates, thus the problem.

However: i've just reinstalled Central Cacti, observed new templates and they got... the templates destroyed. Isn't like:
* the remote pollers overwrite graph templates? is what is the rule for synchronization, hat prevails?...
* why remote pollers password has been set to Cacti Central after I reinstalled the Central Cacti?...

I think I am redo the whole set just in case, it really seems this extra logic for distributing data needs to be reviewed. 
Slightly hijacking this thread, why i am getting: 
FATAL: You are not authorized to use this service 

when trying to access some devices on remote pollers?

**Comment by cigamit** [comment]
Please goto Console > Data Collection > Data Collectors and edit each Data Collector and ensure that the hostname is known on the Remote Data Collectors as saved is at least the same as the first level of the name saved there.  So, for example:  Remote Data Collector has saved the name 'myhost.mydomain.mynet', when you perform a reverse dns on the remote host from the main Data Collector, it has to return at a minimum 'myhost', or preferably the entirety of the DNS name.

If passing through a proxy, we do look review the name behind the NAT as well as the NAT IP.  Either have to resolve to the name saved in the remote data collector.

**Comment by stefanbrudny** [comment]
Finally I got this solved.

Direct issue was indeed damaged template on central server. But why the template was damaged in the first place? 

1. since there is literally zero documentation on this, I started with building Remote Pollers prior to Central Cacti (why not?). This somehow resulted in some weirdly broken central installation later.
2. all templates on central cacti have been empty for some reason, I could see them fine on remote poller, what was not caught in due time.
3. migration process for tables during installation of the central cacti exploded, which was ignored. I do not know whether this was a reason, too. 
4. 100x more items including completely unsupported hacks and tweaks with pollers to run different spines on same host etc.

Solution was to redo the Central installation and remote pollers on apache, one by one and migrate them to nginx. 

All in all I failed to determine the exact procedure for the failure.
